ProjectLighthouse.Servers.GameServer/Controllers/Resources/PhotosController.cs

```diff
@@ -58,13 +58,16 @@ public async Task<IActionResult> UploadPhoto()
         {
             bool validLevel = false;
             PhotoSlot photoSlot = photo.XmlLevelInfo;
+            Slot? adventureSlot = null;
             if (photoSlot.SlotType is SlotType.Pod or SlotType.Local) photoSlot.SlotId = 0;
             switch (photoSlot.SlotType)
             {
                 case SlotType.User:
                 {
-                    Slot? slot = await this.database.Slots.FirstOrDefaultAsync(s => s.Type == SlotType.User && s.SlotId == photoSlot.SlotId);
+                    // We'll grab the slot by the RootLevel and see what happens from here.
+                    Slot? slot = await this.database.Slots.FirstOrDefaultAsync(s => s.Type == SlotType.User && s.ResourceCollection.Contains(photoSlot.RootLevel));
                     if (slot != null && !string.IsNullOrEmpty(slot.RootLevel)) validLevel = true;
+                    if (slot != null && slot.IsAdventurePlanet) adventureSlot = slot;
```

Comment:
```suggestion
                    if(slot == null) break;
                    
                    if (!string.IsNullOrEmpty(slot.RootLevel)) validLevel = true;
                    if (slot.IsAdventurePlanet) photoSlot.SlotId = slot.SlotId;
```
Since we only use the adventureSlot variable if it's an adventure planet and we explicitly check if adventureSlot was set we can just replace the XmlLevelInfo.SlotId if it's an adventure planet